src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -213,7 +209,8 @@ internal sealed override ImmutableArray<string> NotNullWhenFalseMembers
                    isIterator)
         {
             _property = property;
-            _isAutoPropertyAccessor = isAutoPropertyAccessor;
+            _accessorBodyShouldBeSynthesized = (property.IsAutoProperty && syntax is AccessorDeclarationSyntax { SemicolonToken.RawKind: (int)SyntaxKind.SemicolonToken, ExpressionBody: null, Body: null }) ||
+                                                property is SynthesizedRecordPropertySymbol;
```

Comment:
>property is SynthesizedRecordPropertySymbol [](http://example.com/codeflow?start=48&length=43)

Instead of handling all possible flavors of accessors here, I think it would be better for the consumer to pass ```bodyShouldBeSynthesized``` flag to use (for record accessors we don't even need to try to check syntax). Basically we will have three body-related flags: ```hasBody```, ```hasExpressionBody``` and ```bodyShouldBeSynthesized```. Also, I think we should avoid using ```IsAutoProperty```. Passing a flag, let's say named something like "bodyShouldBeSynthesizedForSemicolonOnly", from property's constructor should be sufficient.